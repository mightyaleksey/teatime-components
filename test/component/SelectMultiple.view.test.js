'use strict';

const SelectMultiple = require('../../component/SelectMultiple');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(SelectMultiple.defaultProps.size).toBe('m');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <SelectMultiple
      name='control'
      options={[]}/>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <SelectMultiple
      className='outer'
      disabled
      hasFixedWidth
      id='controls-id'
      name='control'
      options={[
        {label: 'aa', value: 'aa'},
        {label: 'bb', value: 'bb'},
        {label: 'cc', value: 'cc'},
        {label: 'dd', value: 'dd'},
        {label: 'ee', value: 'ee'},
      ]}
      optionsLimit={4}
      placeholder='__'
      searchable
      searchEmptyText='Nothing :('
      searchEngine='startsWith'
      value={['dd']}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
