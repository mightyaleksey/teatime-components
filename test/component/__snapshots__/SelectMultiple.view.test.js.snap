// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<span
  className="wrapper"
  onChange={undefined}
>
  <input
    disabled={undefined}
    id={undefined}
    name="control"
    type="hidden"
    value=""
  />
  <span
    className="baseline"
  >
    <button
      className="control hasPlaceholder"
      disabled={undefined}
      onBlur={undefined}
      onClick={[Function]}
      onFocus={undefined}
      onKeyDown={[Function]}
      type="button"
    >
      â€”
    </button>
    <i
      className="arrow"
      onClick={[Function]}
    />
  </span>
  <span
    className="menu bottom menu isClosedMenu"
    role="listbox"
  />
</span>
`;

exports[`Renders correctly with defined props 1`] = `
<span
  className="wrapper outer"
  onChange={undefined}
>
  <input
    disabled={true}
    id="controls-id"
    name="control"
    type="hidden"
    value="dd"
  />
  <span
    className="baseline"
  >
    <input
      className="search isPseudoFocusedSearch"
      disabled={true}
      onBlur={undefined}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={undefined}
      onKeyDown={[Function]}
      role="combobox"
      value=""
    />
    <span
      className="control"
      disabled={true}
      onBlur={null}
      onClick={[Function]}
      onFocus={null}
      onKeyDown={[Function]}
      type="button"
    >
      dd
    </span>
    <i
      className="arrow"
      onClick={[Function]}
    />
  </span>
  <span
    className="menu bottom menu isClosedMenu isFixedMenu"
    role="listbox"
  />
</span>
`;
